back-end/tests/test_app.py

```diff
@@ -40,57 +40,62 @@ def client(app):
 
 
 def _set_up(basedir):
-    print(os.getcwd())
+    # print(os.getcwd())
     base_dir = to_unix(basedir)
 
     dataset_dir = to_unix(osp.join(base_dir, 'dataset'))
-    print(dataset_dir)
     dataset_dir_original = to_unix(osp.join(base_dir, 'dataset_o'))
-    os.rename(dataset_dir, dataset_dir_original)
-    os.mkdir(dataset_dir)
-    print("setup > rename " + dataset_dir + " to " + dataset_dir_original)
-    train_dir = to_unix(osp.join(dataset_dir, 'train'))
-    test_dir = to_unix(osp.join(dataset_dir, 'test'))
-    train_dir_original = to_unix(osp.join(dataset_dir_original, 'train'))
-    test_dir_original = to_unix(osp.join(dataset_dir_original, 'test'))
-    os.mkdir(train_dir)
-    os.mkdir(test_dir)
-    for name in os.listdir(train_dir_original):
-        image_dir = to_unix(osp.join(train_dir, name))
-        if len(image_dir.split('.')) > 1:
-            continue
-        os.mkdir(image_dir)
-        image_dir_original = to_unix(osp.join(train_dir_original, name))
-        for i in range(10):
-            image = to_unix(osp.join(image_dir, "{}.JPEG".format(i)))
-            image_original = to_unix(osp.join(image_dir_original, "{}.JPEG".format(i)))
-            shutil.copy2(image_original, image)
-    for name in os.listdir(test_dir_original):
-        image_dir = to_unix(osp.join(test_dir, name))
-        if len(image_dir.split('.')) > 1:
-            continue
-        os.mkdir(image_dir)
-        image_dir_original = to_unix(osp.join(test_dir_original, name))
-        for i in range(10):
-            image = to_unix(osp.join(image_dir, "{}.JPEG".format(i)))
-            image_original = to_unix(osp.join(image_dir_original, "{}.JPEG".format(i)))
-            shutil.copy2(image_original, image)
-    print("setup > copy images into " + dataset_dir)
+    if osp.exists(dataset_dir):
+        os.rename(dataset_dir, dataset_dir_original)
+        os.mkdir(dataset_dir)
+        print("setup > rename " + dataset_dir + " to " + dataset_dir_original)
+        train_dir = to_unix(osp.join(dataset_dir, 'train'))
+        train_dir_original = to_unix(osp.join(dataset_dir_original, 'train'))
+        os.mkdir(train_dir)
+        for name in os.listdir(train_dir_original):
+            image_dir = to_unix(osp.join(train_dir, name))
+            if len(image_dir.split('.')) > 1:
```

Comment:
Fixed.

This was initially meant to avoid any files being recognized as a folder / image class, e.g. to exclude `dataset/test/labels.txt` from being recognized as a class such as `dataset/test/bird`.

It is a bad practice (at least it could be `image_dir.split('/')[-1].split('.')`). I was doing this because I messed up the parent folder in the path (should be `dataset_o/...` but I use `dataset/...`) when using `os.path.isdir()` and thought it didn't work. Now it's fixed by using `os.path` with correct